File: server/models/student.go
package models

import "time"

type Student struct {
	ID                      uint                      `gorm:"primary key;autoIncrement" json:"id"`
	Name                    *string                   `json:"name"`
	Password                *string                   `gorm:"not null" json:"password" binding:"required"`
	Email                   *string                   `gorm:"unique" json:"email" binding:"required"`
	NISN                    string                    `json:"nisn"`
	ExpoToken               *string                   `json:"expo_token"`
	ParentID                uint                      `json:"parent_id"`
	Parent                  Parent                    `gorm:"foreignKey:ParentID"`
	TeacherID               uint                      `json:"teacher_id"`
	OrganizationID          uint                      `json:"org_id"`
	StudentClassID          uint                      `json:"student_class_id"`
	StudentAttendanceDetail []StudentAttendanceDetail `gorm:"foreignKey:StudentID"`
	CreatedAt               time.Time                 `json:"created_at"`
	UpdatedAt               time.Time                 `json:"updated_at"`
	DeletedAt               *time.Time                `gorm:"index" json:"deleted_at,omitempty"`
}

File: client/types/Student.ts
import { Class } from "./Class";
import { Subject } from "./Subject";

export interface Student {
  id: number;
  name: string;
  email: string;
  updatedAt: string;
  createdAt: string;
  parent: {
    id: number;
    name: string;
    email: string;
    updatedAt: string;
    createdAt: string;
    expo_token: string | null;
  };
  is_checked?: boolean;
}

export interface StudentByClassID {
  id: number;
  name: string;
  nisn: string;
}

export interface StudentAttendance {
  id: number;
  student_class_id: number;
  student_class: Class;
  subject_id: number;
  subject: Subject;
  teacher_id: number;
  time: string;
  student_attendance: any;
}

export interface AddStudentAttendancePayload {
  student_class_id: number;
  subject_id: number;
  teacher_id: number;
  time: string;
}
